Add export image functionality

bartvde opened the pull request:
SDK-511

wnordmann commented on examples/export-image/app.jsx:
```diff
@@ -0,0 +1,57 @@
+  const url = 'https://raw.githubusercontent.com/boundlessgeo/ol-mapbox-style/master/example/data/wms.json';
+  store.dispatch(mapActions.setContext({ url }));
+
+  const exportMapImage = (blob) => {
```
A comment here about how this is the function called by the onExportImage prop would be useful

bartvde replied:
good point, will add

wnordmann commented on examples/export-image/app.jsx:
```diff
@@ -0,0 +1,57 @@
+  };
+
+  // place the map on the page.
+  ReactDOM.render(
```
I think the export would be more interesting if there some some data included, even the random points would demo that layers are included in the export

bartvde replied:
there is the WMS USA states overlay, but it's currently not visible due to the center issue that duck just fixed, so should resolve itself

bartvde replied:
I've rebased this branch so the center is now ok and shows the USA states layer for printing

wnordmann commented on src/components/map.jsx:
```diff
@@ -278,6 +278,11 @@ export class Map extends React.Component {
       this.updateInteraction(nextProps.drawing);
     }
 
+    if (nextProps.print && nextProps.print.exportImage) {
```
a comment here about using the canvas api would be helpful

bartvde replied:
will add

bartvde commented:
@wnordmann I've addressed your review with: https://github.com/boundlessgeo/sdk/pull/580/commits/82fb8d3e8458a4f6b43ae58cb9677a2819548313

@duck I will await your review

bartvde commented:
@duck look at the example, once the image is received the toggle is flipped. So pressing the button again would work and result in a new image.

bartvde commented:
@duck I didn't really like that the example code is responsible for flipping the switch though, but couldn't think of anything better, maybe you have other ideas for that

bartvde commented:
@duck I just rebased this, let me know if you need any changes after I replied to your comment

theduckylittle commented:
@bartvde  -- note my handle on here is @theduckylittle ... so someone else is probably really curious as to why they're getting notifications.

bartvde commented:
ha, good catch ;-)

Sent from my iPhone

> On 3 Aug 2017, at 19:57, theduckylittle <notifications@github.com> wrote:
> 
> @bartvde -- note my handle on here is @theduckylittle ... so someone else is probably really curious as to why they're getting notifications.
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub, or mute the thread.
>
